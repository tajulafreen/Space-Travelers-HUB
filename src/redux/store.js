import { configureStore } from "@reduxjs/toolkit";
import rockets from "./rockets/rocketSlice";
import missionsReducer from "./missions/missionSlice";

const store = configureStore({
    reducer: {
        rockets: rockets,
        missions: missionsReducer
    }

})

export default store