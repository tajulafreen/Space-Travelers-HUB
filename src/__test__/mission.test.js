import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import Mission from '../components/missions/mission';
import store from '../redux/store';
import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../redux/missions/missionSlice';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);
describe('Mission component', () => {
  it('renders correctly', () => {
    const mockStore = configureStore({
      reducer: {
        missions: missionsReducer, // Replace with your actual reducer name
      },
    });

    const { container } = render(
      <Provider store={mockStore}>
        <Mission />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});