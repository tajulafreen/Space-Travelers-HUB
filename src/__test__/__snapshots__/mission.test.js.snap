// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component renders correctly 1`] = `
<div>
  <p
    class="loadingIndicator"
  >
    ...loading
  </p>
  <table
    class="missionTable"
  >
    <thead
      class="missionThead"
    >
      <th>
        <h3>
          Mision
        </h3>
      </th>
      <th>
        <h3>
          Description
        </h3>
      </th>
      <th>
        <h3>
          Status
        </h3>
      </th>
      <th />
    </thead>
    <tbody />
  </table>
</div>
`;
